core/src/main/java/com/novoda/noplayer/player/PlayerFactory.java

```diff
@@ -1,40 +1,49 @@
 package com.novoda.noplayer.player;
 
 import android.content.Context;
+import android.os.Handler;
+import android.os.Looper;
 
 import com.novoda.noplayer.Player;
 import com.novoda.noplayer.drm.DrmHandler;
 import com.novoda.noplayer.drm.DrmSessionCreator;
 import com.novoda.noplayer.drm.DrmSessionCreatorFactory;
 import com.novoda.noplayer.drm.DrmType;
-import com.novoda.noplayer.exoplayer.ExoPlayerTwoImpl;
-import com.novoda.noplayer.exoplayer.ExoPlayerTwoImplFactory;
-import com.novoda.noplayer.mediaplayer.AndroidMediaPlayerImpl;
-import com.novoda.noplayer.mediaplayer.AndroidMediaPlayerImplFactory;
+import com.novoda.noplayer.exoplayer.NoPlayerExoPlayerCreator;
+import com.novoda.noplayer.mediaplayer.NoPlayerMediaPlayerCreator;
+import com.novoda.utils.AndroidDeviceVersion;
 
 public class PlayerFactory {
 
     private static final boolean DOWNGRADE_SECURE_DECODER = false;
 
     private final Context context;
     private final PrioritizedPlayerTypes prioritizedPlayerTypes;
-    private final ExoPlayerCreator exoPlayerCreator;
-    private final MediaPlayerCreator mediaPlayerCreator;
+    private final NoPlayerExoPlayerCreator noPlayerExoPlayerCreator;
+    private final NoPlayerMediaPlayerCreator noPlayerMediaPlayerCreator;
     private final DrmSessionCreatorFactory drmSessionCreatorFactory;
 
-    public PlayerFactory(Context context, PrioritizedPlayerTypes prioritizedPlayerTypes) {
-        this(context, prioritizedPlayerTypes, ExoPlayerCreator.newInstance(), MediaPlayerCreator.newInstance(), new DrmSessionCreatorFactory());
+    public static PlayerFactory newInstance(Context context, PrioritizedPlayerTypes prioritizedPlayerTypes) {
+        Handler handler = new Handler(Looper.getMainLooper());
+        DrmSessionCreatorFactory drmSessionCreatorFactory = new DrmSessionCreatorFactory(AndroidDeviceVersion.newInstance(), handler);
+        return new PlayerFactory(
+                context,
+                prioritizedPlayerTypes,
+                NoPlayerExoPlayerCreator.newInstance(handler),
```

Comment:
If you remove the internal indirection for the creation then you can just use `new NoPlayerE....(handler)`